### DISCUSSION TITLE
lmoments as a new fit method?

### DISCUSSION THREAD

**Discussion by kikocorreoso** [georgebv/pyextremes#92] [Ideas]
Hi,

When I have done some EVA using some datasets, not specifically using `pyextremes`, I get that the MLE fit could be poor. In these cases I have used `lmoments3` and parameters make more sense.

Would it be interesting to add a third fit method using `lmoments3` to complement MLE and EMCEE?

https://github.com/Ouranosinc/lmoments3 (see potential license issues).

**Comment by georgebv** [comment]
Thank you for the suggestion, this could be a great addition. I don't know the theory to implement this myself, but I'd gladly review/accept a PR.

**Comment by kikocorreoso** [comment]
This is how I implemented it in scikit-extremes: https://github.com/kikocorreoso/scikit-extremes/blob/master/skextremes/models/classic.py#L440

I will have a look to the internals of `pyextremes` and check the best way to add it. I will come back when ready. This would add a new dependency...

**Reply by georgebv** [reply]
Thank you. It looks like only a subset of scipy distributions is supported, but we have the big 4 (GEV/Gumbel and GPD/Expon) so this should be enough for 99% of scenarios.

https://github.com/Ouranosinc/lmoments3/blob/main/docs/source/distributions.rst

**Comment by acrosby** [comment]
Scipy itself supports specifying "MM" for the Method of Moments approach to fitting a `.rv_continuous` distribution rather than the default "MLE." Is "Lmoments" similar? Is there an easy way to specify `method="MM"` in the `.fit()` call in _pyextremes_ currently?  Seems like I would need to monkeypatch some things.
